components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/IdentityGovernanceServiceImpl.java

```diff
@@ -162,8 +168,17 @@ public void updateConfiguration(String tenantDomain, Map<String, String> configu
                 for (Property property : properties) {
                     if (connectorProperties[j].equals(property.getName())) {
                         configProperties[j] = property;
-                        configProperties[j].setDisplayName(propertyFriendlyNames.get(configProperties[j].getName()));
-                        configProperties[j].setDescription(propertyDescriptions.get(configProperties[j].getName()));
+                        String resourceName = configProperties[j].getName();
+                        configProperties[j].setDisplayName(propertyFriendlyNames.get(resourceName));
+                        configProperties[j].setDescription(propertyDescriptions.get(resourceName));
+
+                        if (metaData != null && metaData.containsKey(resourceName)) {
+
+                            configProperties[j].setType(metaData.get(resourceName).getType());
+                            configProperties[j].setRegex(metaData.get(resourceName).getRegex());
+
+                        }
+
```

Comment:
Shall we remove the unnecessary new lines?

Reply:
Fixed in commit 8032899333352cf8a310623b49856dfb99ec9767